Reader: add vote banner for US election 2020

bluefuton opened the pull request:
#### Changes proposed in this Pull Request

Following on from the recent banner for the NZ election (https://github.com/Automattic/wp-calypso/pull/46377), this PR adds a vote banner at the top of Reader's Following stream for the US election on 3rd Nov, 2020.

<img width="870" alt="Screen Shot 2020-10-20 at 15 10 40" src="https://user-images.githubusercontent.com/17325/96532023-7aaf8280-12e7-11eb-837d-3debe3e8bce8.png">

It also refactors the vote banner into its own component.

#### Testing instructions

As a user in the US, ensure that the vote banner appears at the top of your Following feed: http://calypso.localhost:3000/read.

As a user outside the US, ensure you do not see the banner at all.

clucasrowlands commented:
If this is a banner for the U.S., let's put the month first, day second. CTA text ("How to vote") looks good — although to confirm, what resource or page will the button direct people to?

Not sure when this banner will be live for users, but I've included other variations for "Remember to vote" to consider. However, the existing copy ("Remember to vote") works well!

> Election Day: Tuesday, November 3
> Don't forget to vote.

> Election Day: Tuesday, November 3
> Make a plan to vote.

> Election Day: Tuesday, November 3
> Do you have a plan to vote?

gibrown commented:
Great! Thanks!

- Could we add this to the main My Sites page too since that is now the default?
- Agree on changing the title to "Election Day: Tuesday, November 3"
- How about we plug early voting in the description: "Early voting is open now in most states."
- How about we change the link to do to https://www.vote.org/? That seemed like the best non-partisan UX I could find.

clucasrowlands commented:
> How about we plug early voting in the description: "Early voting is open now in most states."

That's great, especially if we could update the copy again when the day gets closer, and day of?

bluefuton commented:
Thanks for your feedback!

I've made the following changes:
- linked the button to vote.org instead
- show early voting copy before election day
- on election day, rotate through @clucasrowlands' suggestions
- changed date format to be more 🇺🇸 friendly

Early voting:

<img width="896" alt="Screen Shot 2020-10-21 at 11 51 37" src="https://user-images.githubusercontent.com/17325/96652640-d6374a00-1393-11eb-88ac-a3eb6721ce42.png">

Election day:

<img width="872" alt="Screen Shot 2020-10-21 at 11 51 13" src="https://user-images.githubusercontent.com/17325/96652660-e0594880-1393-11eb-88a8-d37ea8485984.png">
<img width="850" alt="Screen Shot 2020-10-21 at 11 51 04" src="https://user-images.githubusercontent.com/17325/96652665-e2230c00-1393-11eb-86f9-17af381d49c7.png">

bluefuton commented:
> Could we add this to the main My Sites page too since that is now the default?

I'd like that, but we'd have to ask a few folks first I think and potentially get a designer involved.

If we can keep this PR focused on shipping it in Reader first, we can explore adding the component to My Sites too.

gibrown reviewed (approved):
LGTM and working. Let's ship it.

gibrown commented:
A couple of follow up details:
- We should probably be translating these messages based on the user's language (vote.org does have multiple languages, though I can't tell if their whole site is translated).
- @davemart-in do you think we should ship this at the top of "My Sites" also? Design feedback/help?

davemart-in commented:
> @davemart-in do you think we should ship this at the top of "My Sites" also? Design feedback/help?

I will defer to @rickybanister on this one.

rickybanister commented:
My only design feedback is that it seems like you're using an alert banner (red left border) with the pink icon and primary action button. Can we make the left border pink to unify it? There's probably a class or attribute to do so. 

Translating is a good idea, but we can probably ship it in english first.

bluefuton commented:
> Can we make the left border pink to unify it? There's probably a class or attribute to do so.

Good spot @rickybanister. There was an old style overriding the banner component, which I've now removed.

I've also added `translate()` calls for all of the copy.

wp-desktop reviewed (dismissed):
WordPress Desktop CI Failure for job "wp-desktop-mac".

@bluefuton please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/930664). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5049347

Thank you @bluefuton for including a screenshot in the description! This is really helpful for our translators.

simison commented:
Hi everyone!

I appreciate the intent here but I don't think this went very well from i18n perspective (cc @Automattic/i18n) — this is very USA-centric right now and shipped to all world, all locales. That can feel alienating to others.

- I see this in Lithuania, Europe, and it doesn't mention USA. We just had elections here so this is extra confusing.
- It's in English while my UI is in Spanish.

<img width="1196" alt="Screenshot 2020-10-27 at 11 42 46" src="https://user-images.githubusercontent.com/87168/97285129-813f7c00-184a-11eb-9751-ed0e4911564d.png">

Could we please hide this in non-US geo, non-EN locales? Here's a code example for geo bit:

https://github.com/Automattic/wp-calypso/blob/e4a567ebede0e557b103ccac95a4d1273da23c6c/client/signup/controller.js#L133-L135

> Translating is a good idea, but we can probably ship it in english first.

@rickybanister did you mean that we can ship to English locales and keep hidden for other locales, while this gets translated?

rickybanister commented:
I see the edit to your comment @simison—it should only be shown to US folks. We decided it was ok to ship it in English to all locales while waiting for translations since English is the official language in the US and it seemed better to get this out there than wait.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5049347).
